[experimental branch] C++ code generated from `PolymodFinalMacro` causes certain compilers to hang

This is very annoying as a Linux user as I'm forced to use clang instead, and even it complains about the generated code
```
./src/polymod/hscript/_internal/PolymodFinalMacro.cpp:180:56: warning: stack nearly exhausted; compilation time may suffer, and crashes due to stack overflow are likely [-Wstack-exhausted]
```

And the generated code in question is obviously problematic, containing a hell of nested functions contained in a humongous 500k character line:

<img width="1112" height="758" alt="Image" src="https://github.com/user-attachments/assets/614b6bf9-324c-4703-9113-c02921cf2afe" />

--- a/polymod/hscript/_internal/PolymodFinalMacro.hx
+++ b/polymod/hscript/_internal/PolymodFinalMacro.hx
@@ -42,6 +42,8 @@ class PolymodFinalMacro
               finals.push(field.name);
             }
 
+            if (finals.length == 0) continue;
+
             var entryData = [
               macro $v{classPath},
               macro $v{finals}

